<?xml version="1.0" encoding="UTF-8"?>
<launch>
<!-- her er innholdet i hver launch-file som til vanlig vil inkluderes - se eksempelfilen "freak.launch" i samme mappe-->
	<rosparam file="$(find tb_robot)/param/pantilt_tilt.yaml" command="load"/>

	<arg name="urdf_file" default="$(find xacro)/xacro --inorder '$(find tb_robot)/robots/m600.xacro'" />
	  <param name="robot_description" command="$(arg urdf_file)" />

	  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
	    <param name="publish_frequency" type="double" value="50.0" />
	  </node>

	<node name="tb_robottf_node" 		 pkg="tb_robot" type="tb_robottf_node"            respawn="true"/>

	<node name="tb_simulate_actuatorresponse_node"  pkg="tb_simulate" type="tb_simulate_actuatorresponse_node"	 respawn="true"/>
	<node name="tb_simulate_djiresponse_node" 			pkg="tb_simulate" type="tb_simulate_djiresponse_node"	 respawn="true"/>

	<node name="tb_cmd_actuate_node"  pkg="tb_cmd" type="tb_cmd_actuate_node"	 respawn="true" output="screen"/>
	<node name="tb_cmd_exequte_node"  pkg="tb_cmd" type="tb_cmd_exequte_node"	 respawn="true" output="screen"/>
	<node name="tb_cmd_setpoint_node" pkg="tb_cmd" type="tb_cmd_setpoint_node" respawn="true" output="screen"/>

	<node name="map_server" pkg="map_server" type="map_server" args="$(find tb_bringup)/maps/map_slottet.yaml" />

	<node pkg="move_base" type="move_base" respawn="true" name="move_base" output="screen">
	    <rosparam file="$(find tb_bringup)/planners/global_costmap_params.yaml"   command="load" ns="global_costmap" />
	    <rosparam file="$(find tb_bringup)/planners/local_costmap_params.yaml"     command="load" ns="local_costmap" />
	    <rosparam file="$(find tb_bringup)/planners/move_base_params.yaml" 		      command="load" />
	    <rosparam file="$(find tb_bringup)/planners/dwa_local_planner_params.yaml" command="load" />
	    <rosparam file="$(find tb_bringup)/planners/global_planner_params.yaml"    command="load" />
	    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
	    <param name="base_global_planner" value="navfn/NavfnROS" />
	 </node>

	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find tb_bringup)/rviz.rviz"/>
</launch>
